MapVisualization added.

JanWehrmann opened the pull request:
MapVisualization is a base class for (grid based)
map visualizations.

It provides functionality to draw the extents of the 
map and to apply a local frame.

saarnold commented on viz/ExtentsRectangle.hpp:
```diff
@@ -31,6 +31,8 @@
 #include <osg/Geometry>
 #include <osg/LineWidth>
 
+#include <eigen3/Eigen/Core>
```
The include should be `#include <Eigen/Core>`

JanWehrmann committed:
Visualizations inherit from MapVisualization.

Removed unnecessary code the functionality of which is
now provided through MapVisualization.
Added code as needed to enable the use of that
functionality.

JanWehrmann committed:
ExtentsRectangle::update added.

Allows to update the vertices of the ExtentsRectangle.
Also changed construction parameters to const references.

saarnold commented on viz/MapVisualization.hpp:
```diff
@@ -0,0 +1,173 @@
+        virtual void updateMainNode(osg::Node* node) = 0;
+
+        // You must override this method to specify how your internal data is updated.
+        virtual void updateDataIntern(const GridType& data) = 0;
```
There is no need to redefine those two pure virtual functions here.

JanWehrmann committed:
MapVisualization added.

Adds a base class for map visualizations
that provides functionality for applying
a local frame and for drawing a
rectangular border around the extents of
the map.
